contributor/04-mechanics-of-contributing.md

```diff
@@ -0,0 +1,125 @@
+Imagine what would be needed for realizing this benefit by just walking by or calling a trusted committer on the project or even in a group call.    
+
+Did you on your way of trying to unblock yourself find that you were missing documentation or found parts of it to be wrong or out-of-date and had to figure this out for yourself now?
+Often projects are happy to receive additions, updates or corrections for their existing documentation as well - you've just found another opportunity to contribute! 
+(Or just politely provide them with feedback on your experience and what would have helped you.)
+
+### Crafting the code
+
+We all have our preferences and opinions on code style, indentation, etc. 
+The home that the host team lives in will have them as well, individual wallpaper, decoration and hangings incl. 
+Try to adapt and match those preferences even if it's not what you would normally do and even if it is not specified in the projects' `CONTRIBUTING.md`.
+If you are unsure you can always ask politely, nevertheless a guest contribution for a feature or bug fix is not the time to introduce a new way of structuring or formatting project code.
+
+## Submitting the pull request
+
+You've completed all the essential work, figured out all the quirks of the problem and the project you are contributing to, the time you've planned for the new feature to be used comes nearer and you want to make sure your contribution gets merged in as fast and smooth as possible?
+
+Here's what you can do to make reviewing and merging as easy as possible for the the trusted committer and the host team.
+(This might actually be pretty similar to what you might already be doing on your own project to get your changes accepted - if that's the case: great, this is going to come natural to you!)
+
+### Testing and automation
+
+The basic point here is to enable the trusted committer to validate the contribution without your presence and to ensure easy maintainability.
+To achieve this do the following: 
+
+- Add tests for your code contribution so that validating the function of your contribution by others works well, even after some time when you work in other projects or might have stopped to contribute to this project.
+  - Often projects will use automated checks against pull request using those tests and the level of code coverage. Try to meet the criteria these test enforce.
+- Many projects will provide project build and validations scripts to enable you to locally test your changes. 
+  - Use those to ensure that your contribution works as well as possible before opening a pull request. 
+  - Having to review defect pull requests with easy-to-fix errors often bugs trusted committers. They will not fix your code but ask you to do so. This might create more round-trips and slow the merge if there's unnecessary errors.
+  - No one's perfect though. Try your best, use the prepared validation scripts if there are any and give it your best shot with a pull request!
```

Comment:
I would add that tests you broke could also be explicitly be highlighted in the pull request comment to ask for help understanding what happened and how to fix it.

Reply:
```suggestion
  - No one's perfect though. Try your best, use the prepared validation scripts if there are any and give it your best shot with a pull request!
  - I your pull request keeps breaking tests and you can't find out why after giving it your best shot try to highlight those tests in the pull request comment, illustrate your current understanding of the problem and ask for help on it.  
```

Added a small sentence on this.